frontend/src/lib/components/button/iconButton/IconButton.svelte

```diff
@@ -0,0 +1,44 @@
+<script lang="ts">
+  import {Icon} from '$lib/components/icon';
+  import {ButtonBase} from '../base';
+  import type {IconButtonProps} from './IconButton.type';
+
+  type $$Props = IconButtonProps;
+
+  export let icon: $$Props['icon'];
+  export let title: $$Props['title'];
+  export let color: $$Props['color'] = 'primary';
+
+  // Merge classes into $$restProps
+  let classes = 'btn-ghost flex items-center justify-center px-10';
+  if (color != null) {
+    classes += ` text-${color}`;
+  }
+  $$restProps.class = `${classes} ${$$restProps.class ?? ''}`;
+</script>
+
+<!--
+@component
+Use for buttons that only have an icon, such as those in the right half of 
+the header.
+
+### Properties
+
+- `href`: The URL to navigate to. If this is not supplied be sure to
+  provde an `on:click` event handler or other way of making the item
+  interactive.
+- `icon`: The name of the icon to display.
+- `title`: The text title of the button.
+- `color`: The color of the icon. @default 'primary'
+- Any valid properties of an `<ButtonBase>`.
+
+### Usage
+
+```tsx
+  <IconButton on:click={showHelp} icon="help" title="Show help"/>
+```
+-->
+
+<ButtonBase aria-label={title} {title} {...$$restProps} on:click>
```

Comment:
I think we do not need to make the `aria-label` equal to the title. I think we can remove the `aria-label` and let the developer decide whether or not they want to pass the aria label.

Reply:
The reason here is that `title` is [not an accessible description](https://www.w3.org/WAI/tutorials/forms/labels/#using-the-title-attribute) for the function of the button, but it does act as a tooltip in many browsers. And because the `IconButton` doesn't contain any text content, we need some label for it.

Reply:
Yes, but the `title` and the `aria-label` are different things and have different purposes. One is targeted at visual users, and the other one is targeted at assistive technology. Therefore, they cannot be the same. Sometimes, you just need to target assistive technology without affecting the visuals.

Reply:
I agree on a general level, but in this very specific case – a button with an icon with no text – I think it's very hard to come up with any real-world situation where anybody would want those two properties to be different 🤔. Both state the function of the button in text to those who cannot understand the icon (`title`) or see it (`aria-label`). Moreover, to my understanding there's no situation when both would be available to the same user.

Anyway, I'm changing the behaviour in the unitary component thus so both can be overriden, but we still provide useful defaults in most cases:

```ts
if (variant === 'icon') {
  $$restProps['aria-label'] ??= text;
  $$restProps['title'] ??= text;
}
```

Reply:
Maybe the default for the icon could be the actual icon name as most of them are quite descriptive.

Reply:
That's a bit problematic, because the names are not localized. What we could do would be to use a localized default name for each icon but those we don't have because we didn't want to bind the `Icon` component to the backend…

In this case, I would just use `text`, which is a required property. I think it's useful to require that whoever uses the component explicitly adds a textual description of the function for accessibility's sake.